src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -72,45 +67,46 @@ public JwtVendor(final Settings settings, final Optional<LongSupplier> timeProvi
      *   PublicKeyUse: SIGN
      *   Encryption Algorithm: HS512
      * */
-    static Tuple<JWK, JWSSigner> createJwkFromSettings(Settings settings) {
+    static Tuple<JWK, JWSSigner> createJwkFromSettings(final Settings settings) {
         final OctetSequenceKey key;
         if (!isKeyNull(settings, "signing_key")) {
-            String signingKey = settings.get("signing_key");
+            final String signingKey = settings.get("signing_key");
             key = new OctetSequenceKey.Builder(Base64.getDecoder().decode(signingKey)).algorithm(JWSAlgorithm.HS512)
                 .keyUse(KeyUse.SIGNATURE)
                 .build();
         } else {
-            Settings jwkSettings = settings.getAsSettings("jwt").getAsSettings("key");
+            final Settings jwkSettings = settings.getAsSettings("jwt").getAsSettings("key");
 
             if (jwkSettings.isEmpty()) {
                 throw new OpenSearchException(
                     "Settings for signing key is missing. Please specify at least the option signing_key with a shared secret."
                 );
             }
 
-            String signingKey = jwkSettings.get("k");
+            final String signingKey = jwkSettings.get("k");
             key = new OctetSequenceKey.Builder(Base64.getDecoder().decode(signingKey)).algorithm(JWSAlgorithm.HS512)
                 .keyUse(KeyUse.SIGNATURE)
                 .build();
         }
 
         try {
             return new Tuple<>(key, new MACSigner(key));
-        } catch (KeyLengthException kle) {
+        } catch (final KeyLengthException kle) {
             throw new OpenSearchException(kle);
         }
     }
 
-    public String createJwt(
-        String issuer,
-        String subject,
-        String audience,
-        Integer expirySeconds,
-        List<String> roles,
-        List<String> backendRoles,
-        boolean includeBackendRoles
+    public ExpiringBearerAuthToken createJwt(
+        final String issuer,
+        final String subject,
+        final String audience,
+        final long requestedExpirySeconds,
```

Comment:
wdyt about `requestedDurationSeconds`?

Initially when I read this variable I thought it may be an epoch time in seconds.

Reply:
Duration is the length of the window of the token from issuance to expiry. i.e. expiry minus issued_at = duration

Expiry is an exact Date where the token is expired

Reply:
I'm not going to fix this nit; we can do a future pass where we align the claims language and the downstream systems, as these names are spanning OpenSearch core + security plugin.